File: common/src/main/java/net/azisaba/azisabaachievements/api/AzisabaAchievementsProviderSetter.java
package net.azisaba.azisabaachievements.api;

import org.jetbrains.annotations.NotNull;

public class AzisabaAchievementsProviderSetter {
    public static void setInstance(@NotNull AzisabaAchievements instance) {
        AzisabaAchievementsProvider.set(instance);
    }
}

File: cli/src/main/kotlin/net/azisaba/azisabaachievements/api/AzisabaAchievementsProviderSetter.kt
package net.azisaba.azisabaachievements.api

import net.azisaba.azisabaachievements.cli.CLIAzisabaAchievements
import org.jetbrains.annotations.Contract

object AzisabaAchievementsProviderSetter {
    fun setInstance(instance: CLIAzisabaAchievements) {
        AzisabaAchievementsProvider.set(instance)
    }

    @Contract(pure = true)
    fun getInstance(): CLIAzisabaAchievements {
        return AzisabaAchievementsProvider.get() as CLIAzisabaAchievements
    }
}
